Sources/ShopifyCheckoutSheetKit/CheckoutViewController.swift

```diff
@@ -32,6 +58,13 @@ public class CheckoutViewController: UINavigationController {
         presentationController?.delegate = rootViewController
     }
 
+    init(checkout url: URL, eventHandlers: EventHandlers) {
+        let rootViewController = CheckoutWebViewController(checkoutURL: url, eventHandlers: eventHandlers)
+        rootViewController.notifyPresented()
+        super.init(rootViewController: rootViewController)
+        presentationController?.delegate = rootViewController
+    }
```

Comment:
If I understand correctly, this overloads the class to allow both `delegate` and `eventHandlers`. Is this for backwards compatibility sake?

Reply:
Yes, it's what I wanted to highlight on [this comment](https://github.com/Shopify/checkout-sheet-kit-swift/pull/336#discussion_r2207030168). There I'm proposing ways to simplify it.